homeassistant/components/alexa/smart_home.py

```diff
@@ -900,6 +903,40 @@ def async_api_set_volume(hass, config, request, entity):
     return api_message(request)
 
 
+@HANDLERS.register(('Alexa.InputController', 'SelectInput'))
+@extract_entity
+@asyncio.coroutine
+def async_api_select_input(hass, config, request, entity):
+    """Process a set input request."""
+    media_input = request[API_PAYLOAD]['input']
+
+    # attempt to map the ALL UPPERCASE payload name to a source
+    source_list = entity.attributes[media_player.ATTR_INPUT_SOURCE_LIST] or []
+    for source in source_list:
+        # response will always be space separated, so format the source in the
+        # most likely way to find a match
+        formatted_source = source.lower().replace('-', ' ').replace('_', ' ')
+        if formatted_source in media_input.lower():
+            media_input = source
+            break
+    else:
+        _LOGGER.warning(
```

Comment:
You should probably also not call a service when this happens?

Reply:
I was a little divided about it. I could return something like an `INVALID_VALUE` error so the error bubbles up to the UI (currently it'll pull it's standard "OK!" 🙄) but there's also the case where a media_player doesn't have a source_list but does support select_source in which case maybe it'll work for some users and worse case nothing happens (and the user gets the annoying/confusing "OK!").

I can be convinced either way and maybe I'm just overthinking it and it's better to just return an error and fix a component if a complaint comes in.

Reply:
I don't like silent errors. I rather not have us call anything.